README.md

```diff
@@ -1,2 +1,153 @@
 # ilib-loctool-mrkdwn
-A loctool plugin that can parse Slack mrkdwn messages in a json file
+
+Ilib loctool plugin to parse and localize the slack-flavored markdown
+known as "mrkdwn" encoded in js files.
+
+## File Format
+
+A slack application can encode its strings as a simple js file with
+the properties being the unique ids of the strings, and the values
+being strings in mrkdwn format.
+
+Example file:
+
+```javascript
+export default messages = {
+    "unique id 1": "This is a *mrkdwn* string!",
+    "unique id 2": "This _is_ also a <http://example.com|mrkdwn string>."
+};
+```
+
+## Escapes
+
+Whenever there is syntax in the mrkdwn that translators should
+not touch, this plugin converts them into xml-like components.
+
+```
+This is _bold_ and *italic* text.
+```
+
+Becomes the string:
+
+```
+This is <c0>bold</c0> and <c1>italic</c1> text.
+```
+
+for the translators.
+
+In this way, translators only have to deal with xml-like syntax
+(which they are probably already familiar with) instead of the
+various methods of marking up text using mrkdwn.
+
+Each component is numbered so that the translators can switch
+them around as appropriate for the grammar of their target
+language.
+
+## Code
+
+Snippets of code are not processed. If there is an inline
+snippet of code, it will be considered part of the surrounding
+text and represented as an XML-like component. The text of
+the inline snippet will be put into the comments of the
+extracted string so that the translator knows what it is.
+
+Example:
+
+```
+There are many instances of `numcount` used in the code.
```

Comment:
Will this work the same for ``` (triple backtick) which is used for a block of code?

Reply:
For this plugin, yes. They are pretty much the same -- unlocalizable stuff in the middle of the string